src/Aspire.Hosting/ApplicationModel/ResourceUrlsCallbackContext.cs

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+namespace Aspire.Hosting.ApplicationModel;
+
+/// <summary>
+/// 
+/// </summary>
+public class ResourceUrlsCallbackContext(DistributedApplicationExecutionContext executionContext, IResource resource, List<ResourceUrlAnnotation>? urls = null, CancellationToken cancellationToken = default)
+{
+    /// <summary>
+    /// Gets the resource this the URLs are associated with.
+    /// </summary>
+    public IResource Resource { get; } = resource;
+
+    /// <summary>
+    /// Gets the URLs associated with the callback context.
+    /// </summary>
+    public List<ResourceUrlAnnotation> Urls { get; } = urls ?? [];
+
+    /// <summary>
+    /// Gets the CancellationToken associated with the callback context.
+    /// </summary>
+    public CancellationToken CancellationToken { get; } = cancellationToken;
+
+    /// <summary>
+    /// An optional logger to use for logging.
+    /// </summary>
+    public ILogger Logger { get; set; } = NullLogger.Instance;
```

Comment:
This feels a bit odd. Does it follow a pattern from other hosting code?

Reply:
Yeah it's the same patten used in the environment callback context but I didn't set it to an instance in the `ApplicationOrchestrator`. Will fix.